meta/autoid/autoid_test.go

```diff
@@ -323,10 +378,36 @@ func (*testSuite) TestUnsignedAutoid(c *C) {
 	lastRemainOne := alloc.End()
 	err = alloc.Rebase(4, alloc.End()-2, false)
 	c.Assert(err, IsNil)
-	min, max, err = alloc.Alloc(4, 5)
+	min, max, err = alloc.Alloc(4, 5, 1, 1)
 	c.Assert(err, IsNil)
 	c.Assert(max-min, Equals, int64(5))
 	c.Assert(min+1, Greater, lastRemainOne)
+
+	// Test increment & offset for unsigned.
+	alloc = autoid.NewAllocator(store, 1, true, autoid.RowIDAllocType)
+	c.Assert(alloc, NotNil)
+	c.Assert(err, IsNil)
+	c.Assert(globalAutoID, Equals, int64(1))
+
+	increment := int64(2)
+	n = math.MaxUint64 - 100
+	offset := int64(n)
+
+	min, max, err = alloc.Alloc(5, 2, increment, offset)
+	c.Assert(err, IsNil)
+	c.Assert(uint64(min), Equals, uint64(math.MaxUint64-101))
+	c.Assert(uint64(max), Equals, uint64(math.MaxUint64-98))
+
+	c.Assert(max-min, Equals, autoid.CalcNeededIDs(int64(uint64(offset)-1), 2, increment, offset, true))
+	firstID := seekToFirstAutoID(min, increment, offset)
+	c.Assert(uint64(firstID), Equals, uint64(math.MaxUint64-100))
+
+}
+
+func seekToFirstAutoID(base, increment, offset int64) int64 {
+	nr := (base + increment - offset) / increment
```

Comment:
This formula appear many times, please use a function to wrap it.

Reply:
Addressed

Reply:
I found used `nr := ...`  four times , please update other usages